test/test_cuda.py

```diff
@@ -1521,6 +1521,31 @@ def test_caching_pinned_memory_multi_gpu(self):
         self.assertEqual(gpu_tensor1[0], 1)
         self.assertEqual(gpu_tensor0[0], 2)
 
+    @skipIfRocm
+    def test_sum_cpu_gpu_mismatch(self):
+        x = torch.randn(20, dtype=torch.float32, device='cuda')
+        y = torch.randn(1, dtype=torch.float32)
+        try:
```

Comment:
Don't use this pattern.

If the torch.sum() should succeed, just use self.assertEqual. If it should fail, use one of the self.assertRaises variants.

Reply:
I can update that.
Right now the test should fail. If we ever added support for cpu_gpu mismatched input/output tensors we'll need to update this test case then.